spark-jobs/src/main/scala/za/co/absa/enceladus/standardization/StandardizationJob.scala

```diff
@@ -424,53 +429,75 @@ object StandardizationJob {
+      }
+      case (Right(_), Left(er)) => throw new ValidationException(errorMessage, Seq(er))
+      case (Left(er), Right(_)) => throw new ValidationException(errorMessage, Seq(er))
+      case (Left(er1), Left(er2)) => throw new ValidationException(errorMessage, Seq(er1,er2))
     }
-    Atum.setAdditionalInfo(s"raw_record_count" -> rawRecordCount.toString)
   }
 
   /**
     * Gets the number of records in raw data by traversing Atum's checkpoints.
     *
     * @return The number of records in a checkpoint corresponding to raw data (if available)
     */
-  private def getRawRecordCountFromCheckpoints: Option[Long] = {
+  def getRawRecordCountFromCheckpoints(checkpoints: immutable.Seq[Checkpoint]): Either[String, Long] = {
+    getFieldRawRecordFromCheckpoints(rawFieldName, checkpoints)
+  }
+
+  /**
+   * Gets the number of records in source data by traversing Atum's checkpoints.
+   *
+   * @return The number of records in a checkpoint corresponding to source data (if available)
+   */
+  def getSourceRecordCountFromCheckpoints(checkpoints: immutable.Seq[Checkpoint]): Either[String, Long] = {
+    getFieldRawRecordFromCheckpoints(sourceFieldName, checkpoints)
+  }
+
+  private def getFieldRawRecordFromCheckpoints(checkpointName: String, checkpoints: immutable.Seq[Checkpoint]) = {
     import za.co.absa.atum.core.Constants._
-    val controlMeasure = Atum.getControlMeasure
-
-    val rawCheckpoint = controlMeasure
-      .checkpoints
-      .find(c => c.name.equalsIgnoreCase("raw") || c.workflowName.equalsIgnoreCase("raw"))
-
-    val measurement = rawCheckpoint.flatMap(chk => {
-      chk.controls.find(m => m.controlType.equalsIgnoreCase(controlTypeRecordCount))
-    })
-
-    measurement.flatMap(m =>
-      try {
-        val rawCount = m.controlValue.toString.toLong
-        // Basic sanity check
-        if (rawCount >= 0) {
-          Some(rawCount)
-        } else {
-          None
-        }
+
+    val checkPoint = checkpoints
+      .find(c => c.name.equalsIgnoreCase(checkpointName) || c.workflowName.equalsIgnoreCase(checkpointName))
+      .toRight(s"Missing $checkpointName checkpoint")
+
+    val measurement: Either[String, Measurement] = checkPoint match {
+      case Right(checkpoint) => {
+        checkpoint.controls.find(m => m.controlType.equalsIgnoreCase(controlTypeRecordCount))
+          .toRight(s"$checkpointName checkpoint does not have a $controlTypeRecordCount control")
       }
-      catch {
-        case NonFatal(_) => None
+      case Left(x) => Left(x)
+    }
+
+    measurement match {
+      case Right(m) =>
+        val trial = Try({
+          val rowCount = m.controlValue.toString.toLong
+          if (rowCount >= 0) rowCount else throw new Exception(s"Negative value")
+        })
+        trial match {
+        case Success(value) => Right(value)
+        case Failure(ex) => Left(s"Wrong $checkpointName $controlTypeRecordCount value: ${ex.getMessage}")
```

Comment:
We need `Long` to sanity check for negative values. Maybe we can workaround bogus values leading to the throw of `.toLong` by extracting a `safeToLong()` method that throws a more specific error? What do you think, guys?

Reply:
Where would this method be located?

Reply:
It's not implemented, but we can put it into `Utils`.

Reply:
I wouldn't say it's criptic. As you can see in the testing suite, for an empty string value there it would something like : Wrong raw $controlTypeRecordCount value: For input string ""